File: trial/java/springboot-backend/src/main/java/dev/nazna/examplespringbackend/infrastructure/datasource/repository/UserRepository.java
package dev.nazna.examplespringbackend.infrastructure.datasource.repository;

import dev.nazna.examplespringbackend.domain.User;
import dev.nazna.examplespringbackend.domain.exception.NotFoundRuntimeException;
import dev.nazna.examplespringbackend.infrastructure.datasource.dao.UserDao;
import dev.nazna.examplespringbackend.infrastructure.entity.UserEntity;
import lombok.RequiredArgsConstructor;
import org.seasar.doma.jdbc.Result;
import org.springframework.core.convert.ConversionService;
import org.springframework.lang.NonNull;
import org.springframework.stereotype.Repository;
import org.springframework.transaction.annotation.Transactional;

import java.util.Optional;

@Repository
@RequiredArgsConstructor
public class UserRepository {

  private final ConversionService conversionService;
  private final UserDao userDao;

  public UserEntity find(@NonNull Long id) {
    return userDao.selectByUserId(id).orElseThrow(NotFoundRuntimeException::new);
  }

  @Transactional
  public User create(@NonNull UserEntity entity) {
    Result<UserEntity> insertResult = userDao.insert(entity);
    UserEntity result = userDao.selectByUserId(insertResult.getEntity().userId());
    return conversionService.convert(result, User.class);
  }

}

File: trial/kotlin/spring-modulith/src/main/kotlin/dev/nazna/examplespringmodulith/user/repository/UserRepository.kt
package dev.nazna.examplespringmodulith.user.repository

import dev.nazna.examplespringmodulith.lib.NotFoundException
import org.springframework.stereotype.Repository
import dev.nazna.examplespringmodulith.user.domain.User as UserModel

@Repository
class UserRepository(private val queries: Queries) {

    fun findUser(id: String): UserModel {
        val entity = queries.findUser(id.toLong()) ?: throw NotFoundException("有効なユーザーが存在しません id=$id")
        return UserModel(entity.id.toString(), entity.name, entity.createdAt, entity.updatedAt)
    }
}
